milestone and contrib guide update (#4593)
sunpy/sunpy: docs/dev_guide/newcomers.rst

Newcomers' Guide
****************

> Welcome to the sunpy newcomers' guide.

Reviewer: ```suggestion
Welcome to the SunPy newcomers' guide.
```

> If you have come across this page, you just might be new to sunpy.

Reviewer: ```suggestion
If you have come across this page, you just might be new to SunPy.
```

We aim to be a comprehensive Python package that allows solar physicists to deep dive in the vast amount of solar data available.

> Firstly, we want to thank you for your interest in contributing to sunpy!

Reviewer: ```suggestion
Firstly, we want to thank you for your interest in contributing to SunPy!
```

> sunpy is an open project that encourages everyone to contribute in any way possible.

Reviewer: ```suggestion
SunPy is an open project that encourages everyone to contribute in any way possible.
```

> The people who help develop or contribute to sunpy are varied in ability and experience with the vast majority being volunteers who dedicate time each week.

Reviewer: ```suggestion
The people who help develop or contribute to SunPy are varied in ability and experience with the vast majority being volunteers who dedicate time each week.
```

We pride ourselves on being a welcoming community and we would love to have you become a part of our community.

> Although this document mainly focuses on how to make contributions to sunpy's code and documentation, there are other ways to get involved with the sunpy community.

Reviewer 2: ```suggestion
Although this document mainly focuses on how to make contributions to sunpy's code and documentation, there are other ways to get involved with the SunPy community.
```

> Although this document mainly focuses on how to make contributions to sunpy's code and documentation, there are other ways to get involved with the sunpy community.

Reviewer: ```suggestion
Although this document mainly focuses on how to make contributions to sunpy's code and documentation, there are other ways to get involved with the SunPy community.
```

> Although this document mainly focuses on how to make contributions to sunpy's code and documentation, there are other ways to get involved with the SunPy community.

Reviewer: ```suggestion
Although this document mainly focuses on how to make contributions to the core sunpy libraries code and documentation, there are other ways to get involved with the SunPy community.
```

If you have any questions, comments or just want to say hello, we have online chat on `Matrix`_ which requires no registration or a `Google Group`_ which you message.

.. _Matrix: https://openastronomy.element.io/#/room/#sunpy:openastronomy.org
.. _Google Group: https://groups.google.com/forum/#!forum/sunpy

> How to Contribute to sunpy

Reviewer: ```suggestion
How to Contribute to SunPy
```

Reporting Issues
^^^^^^^^^^^^^^^^

If you use sunpy and stumble upon a problem, the best way to report it is by opening an `issue`_ on our GitHub issue tracker.
This way we can help you work around the problem and hopefully fix the problem!

You will need to sign into `GitHub`_ to report an issue and if you are not already a member of Github, you will have to join.
Joining GitHub will make it easier to report and track issues in the future.

> If you do not want to join Github, then another way to report your issue is email the sunpy developers list `sunpy-dev@googlegroups.com`_.

Reviewer: ```suggestion
If you do not want to join Github, then another way to report your issue is email the SunPy developers list `sunpy-dev@googlegroups.com`_.
```

.. code:: bash

    $ conda config --add channels conda-forge

>     $ conda create -n sunpy-dev pip

Author: So the main change now is to install pip then pip install sunpy[dev] this way you get the full deps installed from pypi instead of half the deps from conda and then the ones we forgot in the recipe from pypi.

This will create a new conda environment called "sunpy-dev" and install the latest version of SunPy from the conda-forge channel.
The next step is remove the conda version of SunPy and install the development version of SunPy.
This will create a new conda environment called "sunpy-dev" and install the latest version of pip from the conda-forge channel.
The next step is get a developement version of sunpy.
This will require that `git`_ be installed.
If you have a `GitHub`_ account, we suggest that you `fork`_ the `sunpy repository`_ (the fork button is to the top right) and **use that url for the clone step**.
This will make submitting changes easier in the long term for you:

.. warning::

    Do not clone the sunpy repository into ``$HOME/sunpy``. Depending on the operating system this location is used to store downloaded data files.
    This will cause conflicts later on, so the last argument (``sunpy-git``) on the ``git clone`` line will become the local folder name of the cloned repository.

.. note::
    If you have a `GitHub`_ account, we suggest that you `fork`_ the `SunPy repository`_ (the fork button is to the top right) and **use that url for the clone step** below.

>     This will make submitting changes easier in the long term for you:

Author: This is now just part of the docs so to avoid confusion.

    Otherwise you are free to clone to any other location.

.. code:: bash

    $ conda remove sunpy

>     $ git clone https://github.com/sunpy/sunpy.git sunpy-git

Author: The reason this was the sunpy url was to allow people who don't want to create a github account to participate.

But what is important to know is that these run a series of tests to make sure that the changes do not cause any new errors.
We strongly recommend that any code changes you have had, follow the `PEP8`_ style and that you have ran the code locally to make sure any changes do not break any existing code.
We provide an overview on how to run the test suite :ref:`here <testing>`.

> Now we (the sunpy community) can review the code and offer suggestions and once we are happy, we can merge in the pull request.

Reviewer: ```suggestion
Now we (the SunPy community) can review the code and offer suggestions and once we are happy, we can merge in the pull request.
```

You can rename ``my_fix`` to something more relevant.
This way, you still get acknowledged for the work you have achieved.

> Now you can email this patch to either the  `Google Group`_ or `a sunpy contributor`_.

Reviewer: ```suggestion
Now you can email this patch to the  `Google Group`_ .
```

Just remember, if you have any problems get in touch!

.. _commit guide: https://chris.beams.io/posts/git-commit/
.. _GitHub guide: https://guides.github.com/activities/hello-world/
.. _PEP8: https://realpython.com/python-pep8/
.. _Google Group: https://groups.google.com/forum/#!forum/sunpy

> .. _a sunpy contributor: stuart@mumford.me.uk

Reviewer: ```suggestion
```
